cupy/testing/helper.py

```diff
@@ -101,8 +101,13 @@ def test_func(self, *args, **kw):
             # values are negative.
             if _contains_signed_and_unsigned(kw):
                 inds = _make_positive_indices(self, impl, args, kw)
-                cupy_result = cupy.asnumpy(cupy_result)[inds]
-                numpy_result = cupy.asnumpy(numpy_result)[inds]
+                try:
```

Comment:
I have now merged your code and changed

```
if _contains_signed_and_unsigned(kw):
```

to

```
if _contains_signed_and_unsigned(kw) and \
                    cupy_result.dtype in _unsigned_dtypes:
```

Also I wrote the corresponding testcases.